lib/citgm-node-gyp.js

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env node
+var fs = require('fs');
+var child_process = require('child_process');
+
+var marker_filename = process.env['citgm_node_gyp_marker_filename'];
+if (!marker_filename) {
+  process.exit(1);
+}
+var node_gyp_filename = process.env['citgm_node_gyp_bin_filename'];
+if (!node_gyp_filename) {
+  process.exit(1);
+}
+
+var args = process.argv.slice(1);
+if (args.indexOf('build') !== -1 || args.indexOf('rebuild') !== -1) {
+  fs.writeFileSync(marker_filename, 'node-gyp called');
+}
+
+delete process.env['npm_config_node_gyp'];
```

Comment:
Documentation (https://nodejs.org/api/process.html#process_process_env) says "use delete" so I went with that.